src/pip/_internal/commands/completion.py

```diff
@@ -80,6 +84,7 @@ def __init__(self, *args, **kw):
         self.parser.insert_option_group(0, cmd_opts)
 
     def run(self, options, args):
+        #  type: (Values, List[Any]) -> int
```

Comment:
Stumbled upon this today, do you have any idea why `args` is not `List[str]` instead?  Other commands seems to have the `List[Any]` as well.  Edit: For `install`, seemingly `List[str]` works.

Reply:
Since the base class [run](https://github.com/pypa/pip/blob/master/src/pip/_internal/cli/base_command.py#L102) method of the base class command contains `List[Any]`, I though to keep it that way in the `run` method of the derived classes as well.

Reply:
Thanks, I did not see that.  I did some investigation and turns out it's safe for the base `run` to take `args: List[str]` as well, since the input is from `optparse.OptionParser.parse_args` whose second return value is of that type.  Do you want me to go ahead and fix it or do you want to take care of it since your type annotation of commands PRs depends on it?

Reply:
I assume you mean from here? https://github.com/python/typeshed/blob/master/stdlib/2and3/optparse.pyi#L227

I think I can take care of fixing the annotations in `pip._internal.cli.base_command` once the other 3 PRs I have for the files in `pip._internal.cli.commands` get merged, so that we fix the derived classes first, and then the base class.

Reply:
I was worried that we need to fix the base class first in order for the derived ones to work, though I myself confirmed that it works for the derived without any modification (silly me :smile:).  Keep up the good work!